plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -395,31 +510,23 @@ public IpDeployer getIpDeployer(Network network) {
 
     @Override
     public boolean applyFWRules(Network network, List<? extends FirewallRule> rules) throws ResourceUnavailableException {
-        s_logger.debug("Handling applyFWRules for network " + network.getName() + " with " + rules.size() + " FWRules");
-        if (rules != null && rules.size() == 1 && rules.iterator().next().getType().equals(FirewallRuleType.System)) {
+        if (rules.size() == 1 && rules.iterator().next().getType().equals(FirewallRuleType.System)) {
             s_logger.debug("Default ACL added by CS as system is ignored for network " + network.getName() + " with rule " + rules);
```

Comment:
Wrap in a if (s_logger.isDebugEnabled()) { ... } block to avoid unnecessary pressure on the string pool when debug logging is not enabled.